test/integration/builder/transaction.js

```diff
@@ -316,15 +327,13 @@ module.exports = function(knex) {
     });
 
     it('#1694, #1703 it should return connections to pool if acquireConnectionTimeout is triggered', function() {
-      var db = Knex({
-        client: knex.client.driverName,
-        pool: {
-          min: 0,
-          max: 1
-        },
-        connection: knex.client.connectionSettings,
-        acquireConnectionTimeout: 300
-      })
+      const knexConfig = _.clone(knex.client.config);
```

Comment:
You should be overriding the whole .pool or otherwise you will modify knex.client.config.pool (https://runkit.com/embed/8zngcllrbqta)